superset-frontend/src/filters/components/Range/RangeFilterPlugin.tsx

```diff
@@ -186,41 +218,62 @@ export default function RangeFilterPlugin(props: PluginFilterRangeProps) {
       }
 
       return {
-        lower: lowerRaw > min ? lowerRaw : null,
-        upper: upperRaw < max ? upperRaw : null,
+        lower: lowerRaw > Number(transformScale(min)) ? lowerRaw : null,
+        upper: upperRaw < Number(transformScale(max)) ? upperRaw : null,
       };
     },
-    [max, min, enableSingleExactValue],
+    [max, min, transformScale, value, enableSingleExactValue],
   );
 
   const handleAfterChange = useCallback(
     (value: [number, number]): void => {
-      setValue(value);
-      const { lower, upper } = getBounds(value);
+      let val = value;
+      if (value[0] === min && value[1] === max) {
+        // after a filter value reset, make sure it's a transformed value
+        val = [transformScale(value[0]), transformScale(value[1])];
+      }
+      // antd apparently uses the floor value, not the rounded value...?
+      // which causes issues like log(123) = 2.0899
+      if (val[1] === Math.floor(transformScale(max) / stepSize) * stepSize) {
+        val = [val[0], transformScale(max)];
+      }
+      if (val[0] === Math.floor(transformScale(min) / stepSize) * stepSize) {
+        val = [transformScale(min), val[1]];
+      }
+      // value is transformed
+      setValue(val);
+      // lower & upper are transformed
+      const { lower, upper } = getBounds(val);
       setMarks(getMarks(lower, upper));
-
+      // removed Number
       setDataMask({
-        extraFormData: getRangeExtraFormData(col, lower, upper),
+        extraFormData: getRangeExtraFormData(
+          col,
+          inverseScale(lower),
+          inverseScale(upper),
+        ),
         filterState: {
           value: lower !== null || upper !== null ? value : null,
```

Comment:
What does the filterState.value do? The variable "value" is transformed, meaning it could be the log of the actual desired value. So I'd expect that this would need to actually be
`value.map(inverseScale)`
like how it is below for the lower and upper vars. But the queries appear to run properly, and the text labels are correct.... :thinking: